src/mappings/mappingHandlers.ts

```diff
@@ -163,3 +166,81 @@ export async function handleStateChangeEvent(cosmosEvent: CosmosEvent): Promise<
 
   await Promise.allSettled(recordSaves);
 }
+export async function handleBalanceEvent(
+  cosmosEvent: CosmosEvent
+): Promise<void> {
+  const { event } = cosmosEvent;
+
+  const incrementEventTypes = [
+    EVENT_TYPES.COMMISSION,
+    EVENT_TYPES.REWARDS,
+    EVENT_TYPES.PROPOSER_REWARD,
+    EVENT_TYPES.COIN_RECEIVED,
+    EVENT_TYPES.COINBASE,
+  ];
+
+  const decrementEventTypes = [EVENT_TYPES.COIN_SPENT, EVENT_TYPES.BURN];
+
+  let operation: Operation | null = null;
+
+  if (incrementEventTypes.includes(event.type)) {
+    operation = Operation.Increment;
+  } else if (decrementEventTypes.includes(event.type)) {
+    operation = Operation.Decrement;
+  } else {
+    logger.warn(`${event.type} is not a valid balance event.`);
+    return;
+  }
+
+  logger.info(`Event:${event.type}`);
+  logger.info(`Event Data:${JSON.stringify(cosmosEvent.event)}`);
+
+  const balancesKit = balancesEventKit();
+  const data = balancesKit.getData(cosmosEvent);
+  logger.info(`Decoded Data:${JSON.stringify(data)}`);
+
+  const address = balancesKit.getAttributeValue(
+    data,
+    BALANCE_FIELDS[event.type as keyof typeof BALANCE_FIELDS]
+  );
+
+  const transactionAmount = balancesKit.getAttributeValue(
+    data,
+    BALANCE_FIELDS.amount
+  );
+
+  if (!address) {
+    logger.error('Address is missing or invalid.');
+    return;
+  }
+
+  const { isBLDTransaction, amount } =
+    balancesKit.validateBLDTransaction(transactionAmount);
+
+  if (!transactionAmount || !isBLDTransaction) {
+    logger.error(`Amount ${transactionAmount} invalid.`);
+    return;
+  }
+
+  const entryExists = await balancesKit.addressExists(address);
+
+  if (!entryExists) {
+    await balancesKit.createBalancesEntry(address);
+  }
+
+  const formattedAmount = BigInt(Math.round(Number(amount.slice(0, -4))));
+  await balancesKit.updateBalance(address, formattedAmount, operation);
+}
+
+export async function initiateBalancesTable(block: CosmosBlock): Promise<void> {
+  const newBalance = new Balances(
+    'agoric1estsewt6jqsx77pwcxkn5ah0jqgu8rhgflwfdl'
```

Comment:
Let's create a separate JSON file that has initial accounts and balances—one for the mainnet and one for the local chain. In the config, we set one and use it based on the environment the code is running.

Reply:
Sure, I'll update that